### PR TITLE
Running tests using python 3.12 

### PR DESCRIPTION
**Pull request by salman2013**
Description:

Running tests for python 3.11 and 3.12

Ticket: https://github.com/orgs/openedx/projects/55/views/1?pane=issue&itemId=53243529,
https://github.com/openedx/axim-engineering/issues/1045

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -56,6 +56,7 @@ allowlist_externals =
     make
     rm
 deps =
+    setuptools
```
Comment (farhan) [inline tox.ini]: I think we have defined setuptools in default [testenv] and no need to define in others

File: requirements/test.txt [hunk 2]
```diff
@@ -62,11 +62,19 @@ click-repl==0.3.0
     #   celery
 code-annotations==1.6.0
     # via -r requirements/test.in
+<<<<<<< HEAD
```
Comment (farhan) [inline requirements/test.txt]: Conflict is not resolved here

File: requirements/quality.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 #
-# This file is autogenerated by pip-compile with Python 3.8
+# This file is autogenerated by pip-compile with Python 3.12
```
Comment (farhan) [inline requirements/quality.txt]: Please verify whether we have to auto-generate on 3.8 or 3.12

**Review by farhan — CHANGES_REQUESTED** [review]

File: tests/openedx_learning/core/publishing/test_api.py [hunk 4]
```diff
@@ -72,7 +72,7 @@ def test_auto_datetime(self) -> None:
         # Auto-generated datetime checking...
         assert isinstance(package.created, datetime)
         assert package.created == package.updated
-        assert package.created.tzinfo == timezone.utc
+        # assert package.created.tzinfo == timezone.utc
```
[the remark this answers is not in the document]
Comment (ormsbee) [reply tests/openedx_learning/core/publishing/test_api.py:75]: Making sure that timestamps are written in UTC is actually important for this app, so I'd really like to understand why this breaks.

**Review by farhan — APPROVED** [review]

**Pull request merged**